utils/resolve.js

```diff
@@ -158,10 +158,24 @@ function requireResolver(name, sourceFile) {
     tryRequire(name) ||
     tryRequire(path.resolve(getBaseDir(sourceFile), name))
 
-  if (!resolver) {
+  if (!isResolverValid(resolver)) {
     throw new Error(`unable to load resolver "${name}".`)
-  } else {
-    return resolver;
+  }
+
+  return resolver
+}
+
+function isResolverValid(resolver) {
+  if (!resolver) {
+    return false
+  }
+  switch (resolver.interfaceVersion) {
```

Comment:
can this be an if/else instead of a switch?

Reply:
I've used a switch since a switch was already used to decide between versions in the code previously, but I can definitely change if needed.

Reply:
not a blocker, but it'd be nice - switch statements are horrible :-)